src/components/common/ColonyDecisions/DecisionDialog/DecisionDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { defineMessages } from 'react-intl';
+import { string, object, number, InferType } from 'yup';
+import { Editor } from '@tiptap/react';
+
+import Dialog, { DialogProps } from '~shared/Dialog';
+import { HookForm as Form } from '~shared/Fields';
+import { useAppContext } from '~hooks';
+import { getDomainId } from '~utils/domains';
+import {
+  getDecisionFromLocalStorage,
+  setDecisionToLocalStorage,
+} from '~utils/decisions';
+
+import {
+  DecisionTitle,
+  DecisionBody,
+  DecisionControls,
+  DialogHeading,
+} from '../DecisionDialog';
+
+import styles from './DecisionDialog.css';
+
+const displayName = 'common.DecisionDialog';
+
+const MSG = defineMessages({
+  titleRequired: {
+    id: `${displayName}.titleRequired`,
+    defaultMessage: 'Please enter a title',
+  },
+  descriptionRequired: {
+    id: `${displayName}.descriptionRequired`,
+    defaultMessage: 'Please enter a description',
+  },
+});
+
+export interface DecisionDialogProps extends DialogProps {
+  ethDomainId?: number;
+  editor: Editor;
+}
+
+const validationSchema = object()
+  .shape({
+    title: string().required(() => MSG.titleRequired),
+    motionDomainId: number(),
+    description: string().notOneOf(['<p></p>'], () => MSG.descriptionRequired),
+  })
+  .defined();
+
+export type DecisionDialogValues = InferType<typeof validationSchema>;
+
+const DecisionDialog = ({
+  cancel,
+  close,
+  ethDomainId,
+  editor,
+}: DecisionDialogProps) => {
+  const { user } = useAppContext();
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  const domainId = getDomainId(ethDomainId);
+  const walletAddress = user?.walletAddress || '';
+  const draftDecision = getDecisionFromLocalStorage(walletAddress);
+  const content = draftDecision?.description;
+
+  const handleSubmit = (values: DecisionDialogValues) => {
+    setDecisionToLocalStorage(values, walletAddress);
+    navigate(`${pathname}/preview`);
+    close();
+  };
+
+  return (
+    <Dialog cancel={cancel}>
+      <Form<DecisionDialogValues>
+        defaultValues={{
+          motionDomainId: draftDecision?.motionDomainId ?? domainId,
+          title: draftDecision?.title,
+          description: draftDecision?.description || '<p></p>',
```

Comment:
I have added trim to the title. Trimming the description would require me to implement a custom yup method because we're not dealing with plain text. We're storing the decision text as an html string, which when empty defaults to `<p></p>`. Empty spaces thus might look like `<p>  </p>` and so the built in trim would see nothing wrong.

Going to leave as is for the time being.